Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -4816,12 +4816,16 @@ static void handleAnimationActions(Element& element, uint32_t action)
 
 bool WebPage::shouldIgnoreMetaViewport() const
 {
+    if (!m_page)
+        return false;
+
     RefPtr localMainFrame = m_page->localMainFrame();
-    if (auto* mainDocument = localMainFrame ? localMainFrame->document() : nullptr) {
-        auto* loader = mainDocument->loader();
+    if (RefPtr mainDocument = localMainFrame ? localMainFrame->protectedDocument() : nullptr) {
+        RefPtr loader = mainDocument->protectedLoader();
```

Comment:
No need to use the protected functions since you're storing them in a `RefPtr`, it even causes unnecessary churn to use the protected functions here.